### PR TITLE
Fixing several bugs 

### LINKED ISSUE #120: Several bugs when working with CsvDataSource / CsvFileSink
While working with the CsvDataSource and the CsvFileSinks several bugs are occurring: 

- CsvDataSource tries to find an operator even if the provided operator is empty (CsvDataSource: 176)
- CsvDataSource logging is incomplete / not helpful in CsvDataSource:176
- LineInput and NodeInput `equals()` and `hashCode()` as failing due to weird missing implementation of `equals()` and `hashCode()` in `org.locationtech.jts.geom.LineString`. Accordingly, ValidationTools are failing when checking for equality, as no object equality can be guaranteed when reading data. This seems like a issue in the library itself. I handed in an issue for this in the original repo: https://github.com/locationtech/jts/issues/531
- parsing of lines with multiple geoJson elements fails due to greedy regex 
- headline is not checked if it contains a field namend `uuid`. Doing this would prevent reading files without a headline.

Furthermore, it _seems_ that the `CsvFileConnector` is not thread-safe due to the implemented `getOrInitWriter(...)` that might initialize multiple writer concurrently as it is not marked as synchronized. Couldn't reproduce this issue for long time now. Might be worth investigating in the future.

### PR DESCRIPTION
**Pull request by johanneshiry**
Resolves #120

### CODE REVIEW

**Comment by johanneshiry** [comment]
!test

**Comment by johanneshiry** [comment]
!test

**Comment by johanneshiry** [comment]
!TEST

**Comment by johanneshiry** [comment]
!test

File: src/test/groovy/edu/ie3/datamodel/models/input/connector/LineInputTest.groovy [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+				GridTestData.nodeC,
+				GridTestData.nodeD,
+				2,
+				GridTestData.lineTypeInputCtoD,
+				Quantities.getQuantity(3, Units.METRE),
+				geoJsonReader.read(lineString) as LineString,
+				OlmCharacteristicInput.CONSTANT_CHARACTERISTIC)
+
+		expect:
+		// do NOT change this equals to '==' as this yields true even if it's false
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/models/input/connector/LineInputTest.groovy:51]: 👍

File: src/test/groovy/edu/ie3/datamodel/utils/GridAndGeoUtilsTest.groovy [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			new Coordinate(7.411111, 51.49228)
+		]
+		"{ \"type\": \"LineString\", \"coordinates\": [[7.411111, 51.49228],[7.411111, 51.49228],[7.311111, 51.49228],[7.511111, 51.49228]]}" | [
+			new Coordinate(7.311111, 51.49228),
+			new Coordinate(7.411111, 51.49228),
+			new Coordinate(7.511111, 51.49228)
+		]
+	}
+
+	def "The GridAndGeoUtils should only modify a provided Coordinate at least as possible"() {
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/utils/GridAndGeoUtilsTest.groovy:45]:
```suggestion
	def "The GridAndGeoUtils should only modify a provided Coordinate as least as possible"() {
```

File: src/main/java/edu/ie3/datamodel/models/input/graphics/LineGraphicInput.java [hunk 3]
```diff
@@ -41,12 +41,12 @@ public boolean equals(Object o) {
     if (o == null || getClass() != o.getClass()) return false;
     if (!super.equals(o)) return false;
     LineGraphicInput that = (LineGraphicInput) o;
-    return Objects.equals(line, that.line);
+    return getLine().equals(that.getLine());
```
[the remark this answers is not in the document]
Comment (johanneshiry) [reply src/main/java/edu/ie3/datamodel/models/input/graphics/LineGraphicInput.java:44]: you're right - I changed my template for method generation accordingly now

File: src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java [hunk 4]
```diff
@@ -187,15 +187,22 @@ public CsvDataSource(String csvSep, String folderPath, FileNamingStrategy fileNa
...
-            () -> {
-              log.debug(
-                  "Cannot find operator for node with uuid '{}'. Defaulting to 'NO OPERATOR ASSIGNED'.",
-                  operatorUuid);
-              return OperatorInput.NO_OPERATOR_ASSIGNED;
-            });
+      Collection<OperatorInput> operators,
+      String operatorUuid,
+      String entityClassName,
+      String requestEntityUuid) {
```
[the remark this answers is not in the document]
Comment (johanneshiry) [reply src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java:193]: Why? The intention of this method is to _always_ provide an operator and hence use optional correctly. The alternative would be to return an `Optional<OperatorInput>` and then log the issue where you get the returned operator. But there is no sense in returning `Optional<OperatorInput>` as we **have** the default available and don't want the user to decide if he is passing in `null` or something else. 

Furthermore, the main purposes of this method was and is to **enforce** logging in exactly this style (as debug log) without providing the opportunity to the user to do something else.

**Pull request merged**